legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/RelationalCompilerExtension.java

```diff
@@ -847,6 +863,44 @@ public List<Procedure3<SetImplementation, Set<String>, CompileContext>> getExtra
         });
     }
 
+    private void checkForDuplicateJoins(Database srcDatabase, CompileContext context)
+    {
+        List<Warning> warnings = Lists.mutable.empty();
+        if (srcDatabase.joins != null)
+        {
+            FastListMultimap<String, Join> nameToJoins = ListIterate.groupBy(srcDatabase.joins, join -> join.name);
+            nameToJoins.forEachKey(name ->
+            {
+                MutableList<Join> joins = nameToJoins.get(name);
+                if (joins.size() > 1)
+                {
+                    warnings.add(new Warning(joins.getLast().sourceInformation, "Found joins with duplicate names: " + name));
```

Comment:
I think we should add both joins to the warning, so user can easily navigate between them.  The idea of the last one was for a failure but we are not failing the process anymore...

Reply:
Resolved here: https://github.com/finos/legend-engine/pull/3194